src/contractors.jl

```diff
@@ -20,9 +22,11 @@ end
 Multi-variable Newton operator.
 """
 function 𝒩(f::Function, jacobian::Function, X::IntervalBox, α)  # multidimensional Newton operator
-    m = IntervalBox(Interval.(mid(X, α)))
+    m = Interval.(mid(X, α))
     J = jacobian(X)
-
+    if contains_zero(det(J))
```

Comment:
Isn't this very expensive? This should basically come out of the `\` or LU factorization.

Reply:
I cant imagine the determinant is as expensive as LU - its just a fixed number of multiplicaitons and additions without any memory allocations

Reply:
As far as I understand, the determinant does exactly an LU decomposition.

Reply:
(ah, maybe not if this is for small StaticArrays?)

Reply:
https://github.com/JuliaLang/julia/blob/master/stdlib/LinearAlgebra/src/generic.jl#L1538

Reply:
apparently StaticArrays also performs an LU decomposition for matrices larger than 4x4 https://github.com/JuliaArrays/StaticArrays.jl/blob/53b4a0ef015bd937608b57a69b6483fe13305762/src/det.jl

Reply:
The fact that this check is even necessary is disturbing as we overload `\` for intervals with custom methods in

https://github.com/JuliaIntervals/IntervalRootFinding.jl/blob/master/src/linear_eq.jl

Either something is broken there or somehow the wrong method is picked.

Reply:
Maybe line 165 needs to have StaticVector instead of StaticArray?

We should definitely test if this is actually getting called.

Reply:
maybe it makes more sense to call gaussian_elimination explicitely (instead of the `\` operator) to make sure and obvious that no external algorithm should be called here

Reply:
> The fact that this check is even necessary is disturbing as we overload `\` for intervals with custom methods in
> 
> https://github.com/JuliaIntervals/IntervalRootFinding.jl/blob/master/src/linear_eq.jl
> 
> Either something is broken there or somehow the wrong method is picked.

I can confirm that the internal method is not getting called – in fact it is broken for StaticArrays. And fixing the Array / Vector mixup is not sufficient either…

Reply:
Are we just not actually importing \ from Base?